herbert_core/configuration/@config_store/config_store.m

```diff
@@ -446,27 +451,38 @@ function set_config_path(obj,new_path,varargin)
             % major version number greater than or equal to 4. This does
             % not include the case of the Jenkins build server, when the
             % default folder name incorporates the build name.
-            
+
             %if ~exist('new_path','var') % not currently necessary as is
             %                              always called with new_path,
             %                              but to be aware of the futuire
             %    new_path = '';
             %end
-            
+
             [is_jenk, build_name, workspace] = is_jenkins();
-            if is_jenk                
+            if is_jenk
                 % remove all possible folder paths of the build name
                 % to be able to create valid file name.
                 [~,build_name] = fileparts(build_name);
                 obj.config_folder_name_ = ['mprogs_config_',build_name];
-                
+
             else
-                ver_string = herbert_version();
+                % analyze version number and if it exceeds the number
+                % requested, specify other version folder.
+                % To be extended in a future when verison number changes
+                % and config folder either need changes or not depending on
+                % various Horace version(s)
+                ver = herbert_version('-num');
+                if ver >= 400
+                    ver_string = '_v4';
```

Comment:
```suggestion
                    ver_string = num2str(floor(ver/100))
                    ver_string = ['_v', ver_string];
```
Or perhaps better simply:
```suggestion
                    ver_string = num2str(floor(ver))
                    ver_string = ['_v', ver_string];
```

Reply:
this is not what we want. I do not want configuration folder changes with every version. Only when it is necessary and can not be avoided it should change. 
Otherwise, when we go to version 4 and patch release 4.01 comes users will get it not configured again. Extremely annoying.

Reply:
Hence why the former only updates when the major version changes, but we don't have to manually do so.

Reply:
BTW, this is question of large discussion with Toby, so I am leaving this for further consideration.

Reply:
It seems foolish to introduce a change we're going to have to manually update once a year when it's trivial to automate, either that or we have to put it in the release SMG.

Reply:
Its absolutely non-trivial to introduce reasonable automation because we do not know the purpose.  I believe that we do not generally want configuration changing with version and if it changes, autoconfiguration should work. This is specific case of the fundamental changes in the code when **users** may want to run two versions simultaneously. Normally such behavior is depreciated. 

SMG is useful in any case. Unfortunately there are so many such cases. 

I am asking Toby to contribute